src/components/views/elements/SettingsFlag.tsx

```diff
@@ -32,34 +33,52 @@ interface IProps {
     isExplicit?: boolean;
     // XXX: once design replaces all toggles make this the default
     useCheckbox?: boolean;
-    disabled?: boolean;
-    disabledDescription?: string;
     hideIfCannotSet?: boolean;
     onChange?(checked: boolean): void;
 }
 
 interface IState {
     value: boolean;
+    disabled: boolean;
```

Comment:
it'd be nice to give this a doc-comment:

```suggestion
    /** true if `SettingsStore.isEnabled` returned false. */
    disabled: boolean;
```